test/capability-tool-test.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable func-names, no-unused-expressions */
+const fs = require('fs');
+const path = require('path');
+
+const { expect } = require('chai');
+
+const CapabilityTool = require('../lib/capability-tool');
+
+function readFixture(filename) {
+  return JSON.parse(fs.readFileSync(path.normalize(`${__dirname}/fixtures/${filename}`, 'utf8')));
+}
+
+describe('CapabilityTool', function () {
+  beforeEach(function () {
+    const capabilityStatement = readFixture(('valid-capability-statement.json'));
+    this.capabilities = new CapabilityTool(capabilityStatement);
+  });
+
+  describe('#supportFor', () => {
+    it('returns false when no arguments are passed', function () {
+      const noArgSupport = this.capabilities.supportFor({});
+
+      expect(noArgSupport).to.be.false;
+    });
+
+    it('returns true when a resource interaction capability is in the capability statement', function () {
+      const patientReadSupport = this.capabilities.supportFor({ resourceType: 'Patient', capabilityType: 'interaction', where: { code: 'read' } });
+
+      expect(patientReadSupport).to.be.true;
+    });
+
+    it('returns false when a resource interaction capability is not in the capability statement', function () {
+      const patientReadSupport = this.capabilities.supportFor({ resourceType: 'Patient', capabilityType: 'interaction', where: { code: 'foo' } });
+
+      expect(patientReadSupport).to.be.false;
+    });
+
+    it('returns true when a conditional create capability is in the capability statement', function () {
+      const conditionalCreateSupport = this.capabilities.supportFor({ resourceType: 'Patient', capabilityType: 'conditionalCreate' });
+
+      expect(conditionalCreateSupport).to.be.true;
+    });
+
+    it('returns true when a specific search param capability is in the capability statement', function () {
+      const birthDateSearchSupport = this.capabilities.supportFor({ resourceType: 'Patient', capabilityType: 'searchParam', where: { name: 'birthdate' } });
+
+      expect(birthDateSearchSupport).to.be.true;
+    });
+
+    it('returns false when a specific search param capability is in the capability statement', function () {
```

Comment:
Should this be *is NOT in*? `searchParam` is an optional field, so I think we should also test that `false` is returned when `searchParam` is not present.